### PR TITLE
fix safetensor saving

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do ?
This PR fixes the `save_model` function for safetensors. It was never tested so it had a couple of problems.

### CODE REVIEW

File: src/accelerate/accelerator.py [hunk 1]
```diff
@@ -2538,7 +2541,10 @@ def save_model(
 
         # Save the model
         for shard_file, shard in shards.items():
-            self.save(shard, os.path.join(save_directory, shard_file))
+            if safe_serialization:
+                safe_save_file(shard, os.path.join(save_directory, shard_file), metadata={"format": "pt"})
```
Comment (muellerzr) [inline src/accelerate/accelerator.py:2545]: Do we know for certain this only happens in `process_index==0`? Otherwise I'd rather see `self.save` be modified (which is really `utils.other.save`) to use `safe_serialize_file` instead.

**Review by muellerzr — COMMENTED** [review]
Thanks! This seems like a good improvement, but we need to be careful on process indexes. Gave a recommendation, or you can directly check `self.is_main_process`

File: src/accelerate/accelerator.py (hunk 1 above)
Comment (SunMarc) [reply src/accelerate/accelerator.py:2545]: I guess it is safer to modify `self.save` instead. Modified !

File: src/accelerate/accelerator.py [hunk 2]
```diff
@@ -2427,7 +2428,7 @@ def save(self, obj, f):
         >>> accelerator.save(arr, "array.pkl")
         ```
         """
-        save(obj, f)
+        save(obj, f, safe_serialization)
```
Comment (BenjaminBossan) [inline src/accelerate/accelerator.py:2431]: Let's pass `safe_serialization` as keyword arg.

File: src/accelerate/accelerator.py [hunk 3]
```diff
@@ -2507,7 +2508,7 @@ def save_model(
                             del state_dict[name]
                             warn_names.add(name)
             if len(warn_names) > 0:
-                logger.warning_once(
+                logger.warning(
```
Comment (BenjaminBossan) [inline src/accelerate/accelerator.py:2511]: Why was this changed?

File: src/accelerate/accelerator.py [hunk 4]
```diff
@@ -2538,7 +2539,7 @@ def save_model(
 
         # Save the model
         for shard_file, shard in shards.items():
-            self.save(shard, os.path.join(save_directory, shard_file))
+            self.save(shard, os.path.join(save_directory, shard_file), safe_serialization)
```
Comment (BenjaminBossan) [inline src/accelerate/accelerator.py:2542]: Let's pass `safe_serialization` as keyword arg.

**Review by BenjaminBossan — APPROVED** [review]
Looks pretty good, thanks for the addition. I would prefer the new argument to passed as keyword argument, as more arguments could be added in the future and we shouldn't rely on order (see #1953). Feel free to merge once this is changed, or to disagree :)

Apart from that, only a small question.

File: src/accelerate/accelerator.py (hunk 3 above)
Comment (SunMarc) [reply src/accelerate/accelerator.py:2511]: it was copy pasted from transformers but logger in accelerate don't have the `warning_once` function

File: src/accelerate/accelerator.py (hunk 2 above)
Comment (SunMarc) [reply src/accelerate/accelerator.py]: sounds good !

File: src/accelerate/accelerator.py (hunk 4 above)
Comment (SunMarc) [reply src/accelerate/accelerator.py]: yes !

**Review by muellerzr — APPROVED** [review]
Great work! :)

**Comment by BenjaminBossan** [comment]
Thanks for implementing the suggestions, LGTM.
